tests/test_searchspace.py

```diff
@@ -187,6 +215,46 @@ def test_invalid_constraint_parameter_combos():
             ],
         )
 
+    # Attempting constraints over parameter sets containing non-numerical discrete
+    # parameters.
+    with pytest.raises(
```

Comment:
I think the first example is sufficient for now (they all pass through the same mechanism internally). When we finally refactor the tests at some point, then we can cleverly parametrize it to avoid maintaining this hard-coded list of test cases. In the long term, there will be an integration test module taking care of that

Reply:
Sounds good! I delete the others and just keep the first one.